### PR TITLE
Updated cmake build, added yatool build

### PR DESCRIPTION
**Pull request by alexv-smirnov**
### Changelog category 

* Documentation (changelog entry is not required)

### CODE REVIEW

File: BUILD.md [hunk 1]
```diff
@@ -1,39 +1,94 @@
 # Building YDB from sources
 
-From this repository you can build YDB Server and YDB CLI (Command Line Interface utility) executables.
+From this repository you can build YDB Server and YDB CLI (Command Line Interface utility) executables. You can build using either Yatool or CMake:
 
-## Build Requirements
+- [Yatool](https://github.com/yandex/yatool) is a primary multilanguage build/test system being used for [YDB development](https://ydb.tech/docs/development/build-ya). You need to use Yatool and follow the [YDB development](https://ydb.tech/docs/development/suggest-change) process to make contributions to the YDB project.
```
Comment (blinkov) [inline BUILD.md:5]:
```suggestion
- [Yatool](https://github.com/yandex/yatool) is the primary multilanguage build/test system being used to [build and test YDB](https://ydb.tech/docs/en/development/build-ya). You need to use Yatool and follow the [YDB development process](https://ydb.tech/docs/en/development/suggest-change) to make contributions to the YDB project.
```

File: BUILD.md [hunk 2]
```diff
@@ -1,39 +1,94 @@
...
+- [CMake](https://en.wikipedia.org/wiki/CMake) is a secondary build system available, with CMakelists.txt automatically pregenerated by Yatool. You can build any binary atrifact from both trunk and release versions of YDB using CMake.
```
Comment (blinkov) [inline BUILD.md:6]:
```suggestion
- [CMake](https://en.wikipedia.org/wiki/CMake) is a secondary build system available, with CMakelists.txt files automatically pregenerated by Yatool. You can build any binary artifact from both `main` and stable release versions of YDB using CMake.
```

File: BUILD.md [hunk 3]
```diff
@@ -1,39 +1,94 @@
...
+
+```bash
+git checkout 23.2.12
+```
+
+### Check out the latest stable YDB CLI version for build
+
+To build a latest stable version of a YDB CLI, check out the latest stable CLI tag from the repository. To do so, visit the https://github.com/ydb-platform/ydb/releases page, scroll down to the top-most 'YDB CLI' release, and use the provided 'tag' value in the `git checkout <tag>` command.
+
+For example, at the time of writing the latest YDB CLI release was [YDB CLI 2.5.0](https://github.com/ydb-platform/ydb/releases/tag/CLI_2.5.0) with a tag `CLI_2.5.0`, so the command looks like that:
```
Comment (blinkov) [inline BUILD.md:55]:
```suggestion
For example, at the time of writing the latest YDB CLI release was [YDB CLI 2.5.0](https://github.com/ydb-platform/ydb/releases/tag/CLI_2.5.0) with a tag `CLI_2.5.0`, so the checkout command looks like this:
```

File: BUILD.md [hunk 4]
```diff
@@ -1,39 +1,94 @@
...
+
+```bash
+git checkout CLI_2.5.0
+```
+
+## Build using Yatool
+
+You can use native Yatool build for x86_64 under Linux Ubuntu 18+.
```
Comment (blinkov) [inline BUILD.md:63]:
```suggestion
You can use the native Yatool build for x86_64 under Ubuntu Linux 18+.
```

File: BUILD.md [hunk 5]
```diff
@@ -1,39 +1,94 @@
...
+
+Run `ya make` from the repository root, followed by an optional relative path to the build target. By default, `ya make` compiles a `debug` configuration, while you can choose to have `release` or `relwithdebinfo`. The latter is recommended for faster builds, as is leverages the YDB remote build cache.
```
Comment (blinkov) [inline BUILD.md:65]: it'd be useful to clarify what's the difference between the mentioned build modes and when to use which

File: BUILD.md [hunk 6]
```diff
@@ -1,39 +1,94 @@
...
+
+To perform build, Yatool downloads and caches relevant toolchains from the YDB S3 storage, so you do not need to install anything on the build machine.
```
Comment (blinkov) [inline BUILD.md:67]:
```suggestion
To build a binary, Yatool downloads and caches relevant toolchains from the YDB S3 storage, so you do not need to install anything on the build machine.
```

File: BUILD.md (hunk 5 above)
Comment (alexv-smirnov) [reply BUILD.md:74]: Can we leave it for later fixes?

**Review by fomichev3000 — APPROVED** [review]
ОК

**Pull request merged**